extensions-core/hdfs-storage/src/main/java/io/druid/storage/hdfs/HdfsDataSegmentPusher.java

```diff
@@ -53,7 +53,8 @@
 
   private final Configuration hadoopConfig;
   private final ObjectMapper jsonMapper;
-  private final String fullyQualifiedStorageDirectory;
+  private final Path storageDir;
+  private String fullyQualifiedStorageDirectory;
```

Comment:
This should be `volatile` since the object is shared amongst multiple threads.

Reply:
Went with the memoize approach